torch/_dynamo/symbolic_convert.py

```diff
@@ -2391,6 +2412,12 @@ def get_trace_call_log_str():
             msg = f"SKIPPED INLINING {code}: {e}"
             log.debug(msg)
             raise Unsupported(msg) from e
+        except exc.InlinedUserStopIteration:
+            if not allow_stopiteration or not isinstance(
+                tracer, InliningGeneratorInstructionTranslator
+            ):
+                raise
+            tracer.symbolic_result = ConstantVariable(None)
```

Comment:
How come?

Reply:
Well, we need to handle inlining the zip case as a polyfill (and other polyfills) where we allow escaping if any StopIteration is emitted by the user code.